File: EmsBMSReports/DlgSelectBMSAlarmGroup.vb
Imports System.Windows.Forms
Imports System.Data.Odbc
Imports MactusReportLib.MactusReportLib

Public Class DlgSelectBMSAlarmGroup

    Public m_sGroupName As String
    Private Sub DlgSelectBMSAlarmGroup_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        SynchronizeAlarmGroupName()
        Dim sQuery As String = "SELECT * FROM tbl_datagroups WHERE grouptype<>0 ORDER BY groupid"
        Dim oReader As OdbcDataReader
        Try
            Using oConnection As New OdbcConnection(g_sConString)
                oConnection.Open()
                Dim oCmd As New OdbcCommand(sQuery, oConnection)
                oReader = oCmd.ExecuteReader()
                While oReader.Read()
                    oGroup.Items.Add(oReader("groupname"))
                End While
                oConnection.Close()
            End Using
        Catch ex As Exception

        End Try
    End Sub

    Private Sub oGroup_SelectedIndexChanged(sender As Object, e As EventArgs) Handles oGroup.SelectedIndexChanged
        Try
            m_sGroupName = oGroup.SelectedItem.ToString
        Catch ex As Exception
            m_sGroupName = ""
        End Try
        If m_sGroupName.Length > 0 Then

        End If

    End Sub
End Class

File: EmsBMSReports/DlgSelectBMSAlarmGroup.cs
using System;
using System.Data.Odbc;
using static MactusReportLib.MactusReportLib;

namespace EmsBMSReports
{

    public partial class DlgSelectBMSAlarmGroup
    {

        public string m_sGroupName;

        public DlgSelectBMSAlarmGroup()
        {
            InitializeComponent();
        }
        private void DlgSelectBMSAlarmGroup_Load(object sender, EventArgs e)
        {
            SynchronizeAlarmGroupName();
            string sQuery = "SELECT * FROM tbl_datagroups WHERE grouptype<>0 ORDER BY groupid";
            OdbcDataReader oReader;
            try
            {
                using (var oConnection = new OdbcConnection(g_sConString))
                {
                    oConnection.Open();
                    var oCmd = new OdbcCommand(sQuery, oConnection);
                    oReader = oCmd.ExecuteReader();
                    while (oReader.Read())
                        oGroup.Items.Add(oReader["groupname"]);
                    oConnection.Close();
                }
            }
            catch (Exception)
            {

            }
        }

        private void oGroup_SelectedIndexChanged(object sender, EventArgs e)
        {
            try
            {
                m_sGroupName = oGroup.SelectedItem.ToString();
            }
            catch (Exception )
            {
                m_sGroupName = "";
            }
            if (m_sGroupName.Length > 0)
            {

            }

        }
    }
}